File: 刷题/cpp/1138. 字母板上的路径.cpp
class Solution {
public:
    string alphabetBoardPath(string target) {
        string res;
        int x=0,y=0;
        for(auto ch:target){
            int tx=(ch-'a')/5;
            int ty=(ch-'a')%5;
            if(tx<x){
                res.append(x-tx,'U');
            }
            if(ty<y){
                res.append(y-ty,'L');
            }
            if(ty>y){
                res.append(ty-y,'R');
            }
            if(tx>x){
                res.append(tx-x,'D');
            }
            res.push_back('!');
            x=tx,y=ty;
        }
        return res;
    }
};

File: 刷题/golang/1138. 字母板上的路径.go
func alphabetBoardPath(target string) string {
    x,y:=0,0
    res:=[]byte{}
    for _,ch:=range []rune(target){
        tx:=int(ch-'a')/5
        ty:=int(ch-'a')%5
        if tx<x{
            for j:=0;j<x-tx;j++{
                res=append(res,'U')
            }
        }
        if ty<y {
            for j:=0;j<y-ty;j++{
                res=append(res,'L')
            }
        }
        if tx>x{
            for j:=0;j<tx-x;j++{
                res=append(res,'D')
            }
        }
        if ty>y {
            for j:=0;j<ty-y;j++{
                res=append(res,'R')
            }
        }
        res=append(res,'!')
        x,y=tx,ty
    }
    return string(res)
}